File: Cpp/Pascal's Triangle II.cpp
#include<iostream>
#include<vector>

vector<int> getRow(int rowIndex)
    {
        vector<int> st1;
        vector<int> st2;
        st1.push_back(1);
        for(int i=0;i<rowIndex;i++)
            {
                st2.push_back(1);
                int cnt;
                for(int j=0;j<st1.size()-1;j++)
                    {
                        cnt=st1[j]+st1[j+1];
                        st2.push_back(cnt);
                    }
                st2.push_back(1);
                st1=st2;
                st2.clear();
            }
        return st1;
    }

File: Python/Pascal's Triangle II.py
#!/usr/bin/env python

def getRow(self, rowIndex):
    seq = [1]
    if rowIndex == 0:
        return seq
    for i in range(rowIndex+1):
        cnt = []
        for j in range(i+1):
            if j == 0:
                cnt.append(seq[j])
            elif j == i:
                cnt.append(seq[j-1])
            else:
                cnt.append(seq[j] + seq[j-1])
        seq = cnt
    return seq